### DISCUSSION TITLE
MKS Wifi TFT siempre permanece en estado "Desconectado"

### DISCUSSION THREAD

**Discussion by ghost** [luc-github/ESP3D#1050] [F.A.Q] [answered]
Tengo una pantalla MKS TFT28 v3.0, y el modulo wifi que le tengo puesto es el Esp8266-12F extraído de un Wemos D1 Mini, y he seguido este diagrama con sus resistencias y todo:

![oxbJScUZCYqy8RZEqNDTFmZSllUTr1Di5KBgTHBD](https://github.com/user-attachments/assets/5340bce9-2d90-437f-9542-203ab37af429)

En el archivo "mks_config", en el apartado Wifi, tengo la siguiente configuración:
 
#wifi model(MKS WIFI TFT:0;  MKS HLK-WIFI:1)
>cfg_wifi_type:0

#wifi mode(0:sta;1:ap)
>cfg_wifi_mode:0

#wifi name and password
>cfg_wifi_ap_name:Nombre de mi red
>cfg_wifi_key_code:Contraseña de mi red

#0:disable cloud service 1:enable cloud service
>cfg_cloud_enable:0
#cloud service adress
>cfg_wifi_cloud_host:www.baizhongyun.cn
#cloud service port
>cfg_cloud_port:10086

#Dynamic access IP
>cfg_ip_dhcp_flag:1

#IP/mask/gateway
>cfg_ip_address:192.168.3.100
>cfg_ip_mask:255.255.255.0
>cfg_ip_gate:192.168.3.1

Los ajustes que pongo en el mks_config.txt, se aplican y el Esp8266-12F, se conecta a mi red wifi sin problema, en la pantalla MKS, me sale la dirección IP que le ha asignado el router, en el navegado pongo la dirección IP, y me sale la interfaz del Esp8266-12F:

![Captura de pantalla 2024-09-16 223921](https://github.com/user-attachments/assets/adcd9db1-82bd-4535-8222-09804e288b41)

Pero el problema es que el estado de conexión, cambia de "Excepción" a "Desconectado" nunca sale "Conectado" y no puedo conectarme a mi impresora incluso estado el Esp8266-12F conectado a la mi red.

He estado investigando, y el "MKS WIFI TFT v1.0 (que no tengo comprado)", lleva el Esp8266-12S pero yo le he puesto el Esp8266-12F (Que si tengo) ya que según lo que he investigado, son prácticamente los mismos módulos al estar basados en el mismo chip (Misma memoria, Pinout, etc...).

He probado tanto el modo "STA" como el modo "AP" (conectándome a la red wifi que el modulo genera) pero nunca aparece "Conectado" y no soy capaz controlar la impresora ni nada.

He probado a cargarle el Firmware varias veces, este se carga sin problema pero sigue igual.

Resumen: Cambia de  "Excepción" a "Desconectado", nunca sale "Conectado" y no tengo control ninguno.

¿Alguna idea de que puede ser y el porque la MKS TFT28 V3.0 reconoce bien el modulo pero no quiere funcionar?

Gracias de antemano.

**Comment by luc-github** [comment]
What version of ESP3D are you using?

**Reply by ghost** [reply]
Yo no estoy usando ESP3D, solamente he realizado la configuración que indica en el [GitHub oficial de Makerbase ](https://github.com/makerbase-mks/MKS-TFT), en la raíz de la tarjeta SD, he puesto los archivos necesarios para mi pantalla:
-mks_font
-mks_pic
-mks_config.txt
-mkstft28.bin
-MksWifi.bin

Todo se actualiza correctamente quedando los archivos de la siguiente manera en la SD:
-bak_font
-bak_pic
-CONFIG
-MKS_WIFI_CUR
-MKSTFT28

Por lo cual la pantalla y el modulo Wifi, se actualizan correctamente.

Según el [video oficial de Makerbase](https://www.youtube.com/watch?v=D9mtISAw9FQ) con solo hacer esto ya estaría funcionando, pero nada.

¿Hay algo que me falta por hacer o hay que hacer algún paso extra en la interfaz web que genera el Esp8266 para que funcione?

**Reply by luc-github** [reply, marked as answer]
This is ESP3D repository you should ask such support in mks repository, 
HKL-Wifi module use a FW that use AT commands
If you do not put any FW on your esp board I do not sée how you expect it works

**Reply by ghost** [reply]
Vale, pues entonces, ¿el "Firmware" que la pantalla le carga al esp al momento de actualizar, es uno básico para cargar el definitivo? y si es así, ¿cómo le cargo el Firmware definitivo? ¿A través de la interfaz web o de manera externa con un programador?

Perdón por si hago preguntas obvias pero soy nuevo en esto y últimamente ando bastante perdido.
